src/data/cam_thermo.F90

```diff
@@ -1633,12 +1666,12 @@ subroutine get_hydrostatic_energy_1hd(tracer, pdel, cp_or_cv, U, V, T,     &
             species_liq_idx(:) = thermodynamic_active_species_liq_idx_dycore(:)
             species_ice_idx(:) = thermodynamic_active_species_ice_idx_dycore(:)
          else
-            species_idx(:) = thermodynamic_active_species_idx(:)
+            species_idx(:) = thermodynamic_active_species_idx(1:)
```

Comment:
Is "1:" needed (instead of ":")?

Reply:
Thanks for noticing! It appears that SIMA allocates `thermodynamic_active_species_idx` as starting from 0, which is different from CAM:
```fortran
      allocate(thermodynamic_active_species_idx(0:num_advected), stat=ierr)
      call check_allocate(ierr, subname,'thermodynamic_active_species_idx(num_advected)', &
                          file=__FILE__, line=__LINE__)
```

I am not sure what the 0-index is used here (and why it is not present in CAM). git line history shows @peverwhee introducing this (https://github.com/ESCOMP/CAM-SIMA/commit/00f442373289af489cc4caf612b9192f1b636898) for 
> fix indexing error
so I assume this is intentional, but I had to subset the indices for `(1:)` here and a few other locations elsewhere to account for this. Perhaps @peverwhee could chime in on the purpose of this change? Is it OK if I subset the indices with `(1:)` here? I saw in the other loop locations using the indices and they all started with `1` or `dry_air_species_num+1`.

Reply:
I believe that what @jimmielin is correct here (based on my understanding and chatting with @nusbaume). I think this was a bug!